tests/cypress/e2e/form/question_types/dropdown.cy.js

```diff
@@ -0,0 +1,369 @@
+
+        cy.get('@question').find('.multiple-preview-dropdown select').invoke('attr', 'id').then((id) => {
+            cy.get(`#select2-${id}-results`).findByRole('option', { name: 'Option 3' }).should('exist').click();
+        });
+
+        // Check selected options and option labels
+        checkSelectedOptions([1, 2], true);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"], true);
+
+        // Save the form (force is required because the button is hidden by a toast message)
+        cy.findByRole("button", { name: "Save" }).click({ force: true });
+
+        // Reload the page to check if the options are still selected
+        cy.reload();
+
+        // Update the question alias
+        cy.findByRole("option", { name: /New question|Test dropdown question/ }).should('exist').as('question');
+
+        // Click on the question
+        cy.get("@question").click('top');
+
+        // Check selected options and option labels
+        checkSelectedOptions([1, 2], true);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"], true);
+    });
+
+    it('test transferring options from simple to multiple and vice versa', () => {
+        // Add new options
+        cy.findByRole("textbox", { name: "Selectable option" }).type("Option 1");
+        cy.findAllByRole("textbox", { name: "Selectable option" }).eq(1).type("Option 2");
+        cy.findAllByRole("textbox", { name: "Selectable option" }).eq(2).type("Option 3");
+
+        // Select the first option
+        cy.findByRole("combobox", { name: "-----" }).closest(".single-preview-dropdown").find("select").select("Option 1", { force: true });
+
+        // Check selected options and option labels
+        checkSelectedOptions([0]);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"]);
+
+        // Change the question type to "Dropdown (multiple)"
+        cy.findByRole("checkbox", { name: "Allow multiple options" }).check();
+
+        // Check selected options and option labels
+        checkSelectedOptions([0], true);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"], true);
+
+        // Select the second option
+        cy.get('@question').find(".multiple-preview-dropdown span.select2-selection--multiple").click();
+
+        cy.get('@question').find('.multiple-preview-dropdown select').invoke('attr', 'id').then((id) => {
+            cy.get(`#select2-${id}-results`).findByRole('option', { name: 'Option 2' }).should('exist').click();
+        });
+
+        // Check selected options and option labels
+        checkSelectedOptions([0, 1], true);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"], true);
+
+        // Change the question type to "Dropdown (simple)"
+        cy.findByRole("checkbox", { name: "Allow multiple options" }).uncheck();
+
+        // Check selected options and option labels
+        checkSelectedOptions([1]);
+        checkOptionLabels(["Option 1", "Option 2", "Option 3"]);
```

Comment:
Not a blocker, but, IMHO, when we switch from multiple to single value dropdonw, we should do this:
- if there are multiple values selected, all should be deselected,
- if there is a unique value selected, it should remain selected.